Lib/test/test_warnings/__init__.py

```diff
@@ -533,6 +533,14 @@ def test_skip_file_prefixes(self):
                 warning_tests.package("prefix02", stacklevel=3)
                 self.assertIn("unittest", w[-1].filename)
 
+    def test_skip_file_prefixes_file_path(self):
+        with warnings_state(self.module):
+            with original_warnings.catch_warnings(record=True,
+                    module=self.module) as w:
+                warning_tests.outer(
+                    "msg", skip_file_prefixes=(warning_tests.__file__, ))
+                self.assertNotEqual(w[-1].filename, warning_tests.__file__)
```

Comment:
```suggestion
            skipped = warning_tests.__file__
            with original_warnings.catch_warnings(
                record=True, module=self.module,
            ) as w:
                warning_tests.outer("msg", skip_file_prefixes=(skipped,))

            self.assertNotEqual(w[-1].filename, skipped)
```

I don't really know `len(w)` but can you also assert the exact `len(w)` as well as that `w[-1] == w[len(w) - 1]`? thanks in advance!

Reply:
Thanks for the review. I'll accept your changes.

> I don't really know len(w) but can you also assert the exact len(w) as well as that w[-1] == w[len(w) - 1]? thanks in advance!

I've referred [exist code](https://github.com/python/cpython/blob/556dc9b8a78bad296513221f3f414a3f8fd0ae70/Lib/test/test_warnings/__init__.py#L476-L497). So I've used `w[-1]`. But the exist code seems to indicate last result of warn because of repeated warn.
My test doesn't have repeated calling warn. So I'd like to replace `w[-1]` with `w[0]`.
I’d love to hear your thoughts on it.

Reply:
What you can do is:

```py
self.assertEqual(len(w), 1)
self.assertNotEqual(w[-1].filename, skipped)
```

AFAICT, the